project-root/Assets/Scripts/NavMeshSpawner.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class NavMeshSpawner : MonoBehaviour
+{
+    public GameObject monsterSpawn;
+    Vector3 point;
+    public float time;
+    public bool done;
+    public List<GameObject> monsters = new List<GameObject>();
+    //public int enemyIndex;
+    public int spawned;
+    public int amountToSpawn;
+    public int reSpawnTime;
+
+    public float range = 10f;
+
+    bool RandomPoint(Vector3 center, float range, out Vector3 result)
+    {
+        
+        for (int i = 0; i < 10; i++)
+        {
+            
+            Vector3 randomPoint = center + Random.insideUnitSphere * range;
+            NavMeshHit hit;
+            if (NavMesh.SamplePosition(randomPoint, out hit, 1 , NavMesh.AllAreas))
+            {
+               
+                result = hit.position;
+                
+                return true;
+               
+            }
+        }
+        result = Vector3.zero;
+        return false;
+    }
+
+
+     void Spawn()
+    {
+        
+        Vector3 point;

```

Comment:
Any and all variables that are declared inside of a function should be var. Again, please refer to the style guide when it comes to these things.